Lib/fontTools/subset/__init__.py

```diff
@@ -2556,8 +2556,8 @@ def _closure_glyphs(self, font):
 
 		self.glyphs_emptied = frozenset()
 		if self.options.retain_gids:
-			self.glyphs_emptied = realGlyphs - self.glyphs_retained
-			# TODO Drop empty glyphs at the end of GlyphOrder vector.
+			max_retained_glyph_id = max(font.getGlyphID(g) for g in self.glyphs_retained)
```

Comment:
in https://github.com/fonttools/fonttools/pull/1588, @blueshade7 added a `self.last_retained_order` attribute to the Subsetter which should be the same as his `max_retained_glyph_id` local variable. Have a look

Reply:
@garretrieger if you rebase your branch on origin/master, you should see the lines I'm referring to from #1588 appear just below this block

Reply:
Thanks that simplifies things, rebased and updated to use it.